cmd/main.go

```diff
@@ -93,18 +98,64 @@ func main() {
 			"'endpoints', 'configmaps', 'leases', 'endpointsleases' and 'configmapsleases'")
 	flag.StringVar(&leaderElectionID, "leader-elect-resource-name", "b8b2488c.x-k8s.io",
 		"The name of resource object that is used for locking during leader election. ")
+	flag.StringVar(&configFile, "config", "",
+		"The controller will load its initial configuration from this file. "+
+			"Command-line flags will override any configurations set in this file. "+
+			"Omit this flag to use the default configuration values.")
 
 	opts := zap.Options{
 		Development: true,
 	}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
+	flag.Visit(func(f *flag.Flag) {
+		flagsSet[f.Name] = true
+	})
+
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	options, cfg, err := apply(configFile)
+	if err != nil {
+		setupLog.Error(err, "unable to load the configuration")
+		os.Exit(1)
+	}
+
 	kubeConfig := ctrl.GetConfigOrDie()
-	kubeConfig.QPS = float32(qps)
-	kubeConfig.Burst = burst
+
+	kubeConfig.QPS = *cfg.ClientConnection.QPS
```

Comment:
Can we remove the flags then since we have configuration files. We can mark them as deprecated in the follow up and remove them finally. And no need to handle the fallback here.

Reply:
I agree with you about the deprecation of the flags and can be done in a followup PR.